pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -94,38 +95,47 @@ type AlertmanagerSpec struct {
 	// +optional
 	ImagePullPolicy v1.PullPolicy `json:"imagePullPolicy,omitempty"`
 	// Version the cluster should be on.
+	// +kubebuilder:validation:MaxLength=100
 	// +optional
 	Version string `json:"version,omitempty"`
 	// Tag of Alertmanager container image to be deployed. Defaults to the value of `version`.
 	// Version is ignored if Tag is set.
 	// Deprecated: use 'image' instead. The image tag can be specified as part of the image URL.
 	// +optional
+	//nolint:kubeapilinter // since deprecated
 	Tag string `json:"tag,omitempty"`
 	// SHA of Alertmanager container image to be deployed. Defaults to the value of `version`.
 	// Similar to a tag, but the SHA explicitly deploys an immutable container image.
 	// Version and Tag are ignored if SHA is set.
 	// Deprecated: use 'image' instead. The image digest can be specified as part of the image URL.
 	// +optional
+	//nolint:kubeapilinter // since deprecated
 	SHA string `json:"sha,omitempty"`
 	// Base image that is used to deploy pods, without tag.
 	// Deprecated: use 'image' instead.
 	// +optional
+	//nolint:kubeapilinter // since deprecated
 	BaseImage string `json:"baseImage,omitempty"`
 	// An optional list of references to secrets in the same namespace
 	// to use for pulling prometheus and alertmanager images from registries
 	// see https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+	// +kubebuilder:validation:MaxItems=10
 	// +optional
 	ImagePullSecrets []v1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
 	// Secrets is a list of Secrets in the same namespace as the Alertmanager
 	// object, which shall be mounted into the Alertmanager Pods.
 	// Each Secret is added to the StatefulSet definition as a volume named `secret-<secret-name>`.
 	// The Secrets are mounted into `/etc/alertmanager/secrets/<secret-name>` in the 'alertmanager' container.
+	// +kubebuilder:validation:MaxItems=50
+	// +kubebuilder:validation:items:MaxLength=253
```

Comment:
I think that secret names can't exceed 63 chars. For the max number of secrets, I'm not sure what a reasonable limit would be TBH.